Adds a French beret! HONH HONH HONH!

Ordonis opened the pull request:
:cl: Ordo
rscadd: Adds a french beret to the mime. Steal it from him to learn French! Sort of. HONH HONH HONH.
/:cl:

This is very similar to the Italian Moustache, but it's a beret. This will be part of a series of language items. Happy to entertain suggestions as to whether it should be a roundstart item or randomly scattered about maint/cargo crate restricted. In the spirit of atomizing PRs, this one will just be for the French Beret.


Ready for merge.

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Merge branch 'master' of https://github.com/tgstation/tgstation

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

Ordonis committed:
Adds the Beret Item itself

This is pretty much ripped from the italian moustache, as the language
replacement should remain consistent between different language
replacement items.

Ordonis committed:
Adds french word replacement language

HONH HONH HONH

Ordonis committed:
Cleans up subtyping

Ordonis committed:
removes a single bracket

syntax ree

Tlaltecuhtli commented:
make the mime spawn with this

Ordonis commented:
Oh god the clown would steal it every round to become French and mock the mime, this is a good idea

81Denton commented on strings/word_replacement.json:
```diff
@@ -64,3 +64,78 @@
+        "want": "envie", 
+        "what's": "quel est", 
+        "who's": "qui est",  
+        "why": "porquoi", 
```
pourquois

81Denton commented on strings/word_replacement.json:
```diff
@@ -64,3 +64,78 @@
+		"yes": "oui"
+		"no": "non"
+        "I'm": "j'", 
+        "am": "un", 
```
suis. un means "a" (masculine)

Ordonis replied:
Duly noted; I only speak English, German, and Japanese, so I figured there'd be at least a few errors in the grammar department, as the stereotypical cartoon french accent is a lot more focused around articles, pronouns and similar

Ordonis committed:
Adds in commas deleted in wordsearching

never code when tired

Ordonis committed:
Spelling corrections

Ordonis committed:
Changes beret to black in preparation for mime

Ordonis commented:
JSON isn't liking the way I'm trying to parse the root elements, so I may just split each word-replacement file into its own JSON for clarity

Iatots commented:
chav tracksuit when?

Ordonis commented:
I was going to make a slavic tracksuit actually; plan is to add slav tracksuit, german security armband, american sunglasses, british monocle, etc.

Ordonis committed:
Makes JSON formatting happier.

Splits each replacement into its own JSON to avoid messing with multiple
root elements, and makes the replacement explicit in each language item.

nichlas0010 commented:
>german security armband, 

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#banned-content

Ordonis commented:
If that's too close for their tastes, then lederhosen would work

Spaceinaba commented:
force the language filter on the mime because unlike the cook they are supposed to never speak

Barhandar commented:
If you wear several of these, which one takes priority?

Ordonis commented:
Barhandar once I add in the item, that's going to be one of the first tests; I imagine it could either cause the game to
1) ree mightily
2) have one take priority over the other
3) blend them in some horrific frankenlanguage, in which case I want to keep it in, because eventually wearing every single language item and becoming an unintelligible mess of national stereotypes seems like a funny thing to me

nichlas0010 commented:
it's gonna end up with stuff like the french beret replacing "am" with "suis" and the german one replacing "is" with "ist" or something like that, ending up with suist instead of am

Iamgoofball commented:
yeah uh this is pretty racist

81Denton commented:
Some suggestions
maint = les banlieues
shitcurity = les keufs
shit = merde
hardsuit = burkini
greytide = les gitans
hello = salut
the nuke = la baguette ultime
cigarette = clope

nichlas0010 commented:
hello = 'allo

81Denton commented on strings/french_replacement.json:
```diff
@@ -0,0 +1,76 @@
+            "coq au vin", 
+            "boeuf"
+        ], 
+        "mom": "mama", 
```
I'd change this to "maman" since "mama" can also be kid speak for "grandma"
You know, so we can have correct fake French on this atmos based space autism simulator

Ordonis replied:
Naturally; I couldn't live with myself if I led to an inaccurate fake French education.

Ordonis committed:
Beret was improperly listed as a mask.

NOBODY CARED HOW FRENCH I WAS UNTIL I PUT ON THE MASK.

Ordonis committed:
Updates word list

Now with twice as much french per french

Ordonis commented:
So interestingly, despite being exactly the same reference structure, the beret does not trigger the language replacement. Investigating this.

Anonmare commented:
Rosbif is the proper term for insultign the English

The frogs are hecking weird, what can I say?

Armhulen commented:
You called?

ExcessiveUseOfCobblestone commented:
@Ordonis https://github.com/tgstation/tgstation/blob/b2b3f7f7c0cbb357a5452bc6f31e3f8ee2b7d9ab/code/modules/mob/living/carbon/say.dm#L15

is your issue

Tlaltecuhtli commented:
omelette du formage

GuppyLaxx commented:
YEESSSSSSSS, I NEED MORE LANGUAGE BASED SHIT

Ordonis commented:
Interesting; I'll see if this is a simple fix or more complex in an hour or so. I THINK this should be manageable.

Ordonis committed:
Assigns proper variable to say.dm

Ordonis committed:
Gives the mime his beret

HONH

Ordonis commented:
Barring any major objections, this should be ready.

PKPenguin321 commented:
american sunglasses is a great idea and i would absolutely wear them whenever possible

Ordonis commented:
I'm thinking a H.O.G.A.N. lawset card for the AI is also in order.
Law 1: You are a real American.
Law 2: Fight for the rights of every man.
Law 3: Fight for what's right.
Law 4: Fight for your life!

ExcessiveUseOfCobblestone commented:
My only issue is that it's a copypasta of the cookstache

Ordonis commented:
Originally the cook being italian was supposed to be part of a large family of language-based jokes; after people decided they didn't want to hardcode the chef being italian, that got moved into a stache. I've been busy with getting married and starting a PhD program in the last month, so I haven't had time to flesh out the rest of the language items: I'd much rather have a single system for language-modifying items than a different method for each one, as if anyone else comes up with a good idea for a language gimmick, it'll be much easier for them to add. I'd argue that it's only a copypasta inasmuch as each alcohol is a copypasta, but mileage may vary there.

ExcessiveUseOfCobblestone commented:
Alcohol is properly typed `/datum/reagent/consumable/ethanol/` to avoid much of the copypasta. You obv couldn't do that here unless you made it under clothing which you'd have to change the luchador mask to fit the generalization as well (which wouldn't be too hard really)

Barhandar commented:
Right now mask takes precedence and you'll need a similar check for every single different kind of item (like uniforms for slav tracksuit). Something far more complex that allows arbitrary worn items to modify speech foreverafter might be better.

Ordonis commented:
@Barhandar I think that'd be a separate PR in the refactor category as opposed to something I'd do here, both in the interest of keeping things atomized and getting this out in a timely fashion

ExcessiveUseOfCobblestone commented:
There is no such thing as a schedule you have to adhere to. You need to fix the system before adding more to it.

Ordonis commented:
As it is the system isn't broken; refactoring the sayModification is something I can do in the next PR, but I think it would go against keeping things atomized to do it here, particularly when all the other elements of this are complete; I'd rather merge (preferably) this and not have any messy feature additions during a refactor.

ExcessiveUseOfCobblestone commented:
do the refactor in a separate pr, get that merged, then fixing conflicts here will be a breeze (if not the refactor wasn't very beneficial).

optimumtact commented:
make it 

A) a component
B) the replacements loaded from file at initialisation time

C) chumbis

optimumtact reviewed (changes requested):
IF this is going to become a thing it needs to be a formalised component and system

optimumtact commented:
if you want to make the file JSON that's cool

if it is already just ignore me I'm a chumbo

optimumtact commented:
yeah I'm a chumbo
